src/controllers/open-graph.controller.ts

```diff
@@ -0,0 +1,23 @@
+import { getOpenGraph } from '~/routes/open-graph.route';
+import { createRouter } from './router-factory';
+import og, { ErrorResult } from 'open-graph-scraper';
+import {
+  OpenGraphErrorResultSchema,
+  OpenGraphSuccessResultSchema,
+} from '~/types/open-graph.types';
+
+export const openGraphScrapeRoute = createRouter();
+
+openGraphScrapeRoute.openapi(getOpenGraph, async (c) => {
+  const { url } = c.req.valid('query');
+
+  const data = await og({ url })
+    .then((data) => data)
+    .catch((err: ErrorResult) => err);
+
```

Comment:
tambahin http cache biar ga request ke url yang sama berkali2

```suggestion
// cache for one day
c.header('Cache-Control', `public, max-age=${60 * 60 * 24}`)
```